src/components/messages-view-configurator/MessagesViewConfigurator.tsx

```diff
@@ -14,84 +14,110 @@
  * limitations under the License.
  ***************************************************************************** */
 
-import React, { useRef, useState } from 'react';
-import { useOutsideClickListener } from '../../hooks';
+import React, { useState } from 'react';
+import { observer } from 'mobx-react-lite';
 import { ModalPortal } from '../util/Portal';
-import '../../styles/messages-view-configurator.scss';
 import RulesList from './RulesList';
-import { createStyleSelector } from '../../helpers/styleCreators';
 import BodySortConfig from './BodySortConfig';
+import useTheme, { Theme } from '../../hooks/useTheme';
+import useLocalStorage from '../../hooks/useLocalStorage';
+import TogglerRow from '../filter/row/TogglerRow';
+import { useMessageDisplayRulesStore } from '../../hooks';
+import '../../styles/app-settings.scss';
 
 type Props = {
 	sessions: string[];
 };
 
-const MessageViewConfigurator = ({ sessions }: Props) => {
-	const [isOpen, setIsOpen] = useState(false);
-	const [mode, setMode] = useState<'display-rules' | 'body-sort'>('display-rules');
+export enum ThemeNames {
+	LIGHT = 'light',
+	DARK = 'dark',
+}
 
-	const modalRef = useRef<HTMLDivElement>(null);
-	const buttonRef = useRef<HTMLButtonElement>(null);
+export enum HighlightNames {
+	KEYS = 'keys',
+	VALUES = 'values',
+}
 
-	useOutsideClickListener(modalRef, (e: MouseEvent) => {
-		const isFromAutocomplete = Boolean((e.target as HTMLElement).closest('.rules-autocomplete'));
-		const isFromSelect = Boolean((e.target as HTMLElement).closest('.rules-select-options-list'));
-		if (e.target !== buttonRef.current && !isFromAutocomplete && !isFromSelect) {
-			setIsOpen(false);
-		}
-	});
+const themes: { [k: string]: Theme } = {
```

Comment:
not sure why would we manage theme css since we use scss in this project.

i think it's better just to switch classes on body element, each class would have it's own set of css-variables